src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/inference/SimpleNovelAdjacencyInterpreter.java

```diff
@@ -0,0 +1,214 @@
+                                        tig.getSourceContig().alignmentIntervals.get(1),
+                                        MORE_RELAXED_ALIGNMENT_MIN_MQ, MORE_RELAXED_ALIGNMENT_MIN_LENGTH))
+                        .mapToPair(tig -> {
+                            final ChimericAlignment simpleChimera = ChimericAlignment.extractSimpleChimera(tig,
+                                    referenceSequenceDictionaryBroadcast.getValue());
+                            final byte[] contigSequence = tig.getSourceContig().contigSequence;
+                            final NovelAdjacencyReferenceLocations novelAdjacencyReferenceLocations =
+                                    new NovelAdjacencyReferenceLocations(simpleChimera, contigSequence,
+                                            referenceSequenceDictionaryBroadcast.getValue());
+                            final byte[] altHaplotypeSeq = null; // TODO: 1/21/18 force all subtypes of NovelAdjacencyReferenceLocations to extract alt haplotype
+                            final SimpleNovelAdjacency.NovelAdjacencyAndInferredAltHaptype novelAdjacencyAndInferredAltHaptype =
+                                    new SimpleNovelAdjacency.NovelAdjacencyAndInferredAltHaptype(novelAdjacencyReferenceLocations, altHaplotypeSeq);
+                            return new Tuple2<>(novelAdjacencyAndInferredAltHaptype, simpleChimera);
+                        })
+                        .groupByKey()       // group the same novel adjacency produced by different contigs together
+                        .map(noveltyAndEvidence ->
+                                new SimpleNovelAdjacency(noveltyAndEvidence._1, Lists.newArrayList(noveltyAndEvidence._2)));
+
+        SvDiscoveryUtils.evaluateIntervalsAndNarls(assembledIntervals,
+                simpleNovelAdjacencies.map(SimpleNovelAdjacency::getNovelAdjacencyReferenceLocations).collect(),
+                referenceSequenceDictionaryBroadcast.getValue(), discoverStageArgs, toolLogger);
+
+        return simpleNovelAdjacencies;
+    }
+
+    /**
+     * Main method: given simple novel adjacency
+     *  (that is, affected reference locations, alt haplotype sequence, and chimeric alignment evidence),
+     *  infer type.
+     *
+     * @return the inferred type could be a single entry for simple variants, or a list of two entries with BND mates.
+     */
+    static List<SvType> inferSimpleOrBNDTypesFromNovelAdjacency(final SimpleNovelAdjacency simpleNovelAdjacency,
+                                                                final ReferenceMultiSource reference, final SAMSequenceDictionary referenceDictionary) {
+
+        // based on characteristic of simple chimera, infer type
+        final List<ChimericAlignment> alignmentEvidence = simpleNovelAdjacency.getAlignmentEvidence();
+        final boolean allIndicateSimpleTransLoc = alignmentEvidence.stream().allMatch(ChimericAlignment::isLikelySimpleTranslocation);
```

Comment:
You are doing all this evaluation now but then you probrably will not need most of the based on the if else if else ... structure below. Can these be part of a the conditionals to avoid unecessary calculations?

Reply:
corrected. Thanks!